### ISSUE TITLE
IMethodInterceptor executed twice 

### ISSUE THREAD

**Issue by Elisedlund** [cbeust/testng#824]
The intercept method in IMethodInterceptor is always executed twice. it should probably only be executed once?

simple testcase and simple IMethodInterceptor: 
```
public final class Listener implements IMethodInterceptor {
    @Override
    public List<IMethodInstance> intercept(List<IMethodInstance> methods, ITestContext context) {
        System.out.println("intercept ITestContext: " + context.getName() + " methods: " + methods);
        return methods;
    }
}

public class TestClass {
    @Test
    public void helloTest() throws Exception {
        System.out.println("helloTest()");
    }
}

```
console output: 
```
intercept ITestContext: Test methods: [[MethodInstance method=TestClass.helloTest()[pri:0, instance:TestClass@541d4bce] instance=TestClass@541d4bce]]
intercept ITestContext: Test methods: [[MethodInstance method=TestClass.helloTest()[pri:0, instance:TestClass@541d4bce] instance=TestClass@541d4bce]]
helloTest()
```

**Comment by Elisedlund** [comment]
seems like 6.9.8 does not have this issue.

**Comment by juherr** [comment]
True! I forgot I've already fixed it with #804 and only see your traces x]
Sorry.

Close as duplicate of #154
